src/Common/src/Interop/Windows/kernel32/Interop.GetVolumeInformation.cs

```diff
@@ -11,5 +11,7 @@ internal partial class Kernel32
     {
         [DllImport(Libraries.Kernel32, EntryPoint = "GetVolumeInformationW", CharSet = CharSet.Unicode, SetLastError = true, BestFitMapping = false)]
         internal static extern bool GetVolumeInformation(string drive, [Out]StringBuilder volumeName, int volumeNameBufLen, out int volSerialNumber, out int maxFileNameLen, out int fileSystemFlags, [Out]StringBuilder fileSystemName, int fileSystemNameBufLen);
+
+        internal const uint FILE_SUPPORTS_ENCRYPTION = 0x00020000;
```

Comment:
Would you mind fixing this P/Invoke up while you're here? It's wasteful/expensive and not used correctly in DriveInfo.  The [docs](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getvolumeinformationw) are pretty clear about the maximum buffer lengths. I'd switch this to use `char*` and pass null in the cases where we don't care about volume/filesystem name. In `DriveInfo` we can just stackalloc `MAX_PATH + 1` in each of the methods that use it.

The `volSerialNumber` isn't used and can be passed as null too, so you can turn that into `int*` (`IntPtr` would work too, but since we'd already be using pointers here...).

If you want to follow up `DriveInfo` in a separate change you can just add the `char*` overload in this change and call it with null.

Reply:
@JeremyKuhne i can fix DriveInfo in this PR...but if you think is better to have separate changes is not a problem!Let me know!

Reply:
For clarity you might want to do it in two stages (i.e. add a second DllImport temporarily).

Reply:
Ok @JeremyKuhne  i'll add overload here and fix DriveInfo on other PR! Thank's for review and tips!

Reply:
@JeremyKuhne here the amend for DriveInfo on next PR https://gist.github.com/MarcoRossignoli/cb53c616211e3d3785b4d2f06547ee9f#file-driveinfo-windows-cs-L28:L44